pkgs/os-specific/linux/atop/atop.service.patch

```diff
@@ -1,10 +1,17 @@
+diff --git a/atop.service b/atop.service
+index 6b11dea..f90935f 100644
 --- a/atop.service
 +++ b/atop.service
-@@ -9,5 +9,6 @@
- Environment=LOGPATH=/var/log/atop
+@@ -8,10 +8,11 @@ Environment="LOGOPTS="
+ Environment="LOGINTERVAL=600"
+ Environment="LOGGENERATIONS=28"
+ Environment="LOGPATH=/var/log/atop"
 -EnvironmentFile=/etc/default/atop
 +EnvironmentFile=-/etc/default/atop
+ ExecStartPre=/bin/sh -c 'test -d "${LOGPATH}" || mkdir -p "${LOGPATH}"'
```

Comment:
Doesn't `mkdir -p` silently succeed if the target exists already (so no need to test)? Line 14 in this patch suggests that the same directory is being ensured already.

Reply:
Copied this from the previous patches for atop so not entirely sure the behavior here

Reply:
I don't think previous patches for `atop` contained any instance of `test -d`. And you are adding this line while the same functionality was already added by the same patch before your changes.

Reply:
I think I got it from an older version. Will fix.

Reply:
Oh actually this is in the upstream atop.service file that's why we see it in the patch file:

https://github.com/Atoptool/atop/blob/master/atop.service#L12

Reply:
Ah in this case we should probably drop the local addition of the same thing below, if we are changing the patch anyway